### PR TITLE
Resolve non-js package importing

### PR DESCRIPTION
**Pull request by shirohana**
Belongs to issue https://github.com/nuxt/nuxt.js/issues/375#issuecomment-317099812

### CODE REVIEW

File: lib/builder/builder.js [hunk 1]
```diff
@@ -242,6 +242,7 @@ export default class Builder extends Tapable {
           hash,
           r,
           wp,
+          resolveModule,
```
Comment (pi0) [inline lib/builder/builder.js]: For resolving modules there is already a helper function [resolvePath](https://github.com/nuxt/nuxt.js/blob/dev/lib/core/nuxt.js#L101) in code doing same. So we may change this to:
```js
resolvePath: this.nuxt. resolvePath.bind(this.nuxt)
```

File: lib/app/App.vue [hunk 2]
```diff
@@ -8,7 +8,7 @@
 <script>
 <% if (loading) { %>import NuxtLoading from '<%= (typeof loading === "string" ? loading : "./components/nuxt-loading.vue") %>'<% } %>
 <% css.forEach(function (c) { %>
-import '<%= wp(c.src || c) %>'
+import '<%= resolveModule(c.src || c) %>'
```
Comment (pi0) [inline lib/app/App.vue]: We may also wrap this path with `relativeToBuild`

**Comment by shirohana** [comment]
oops, I see இωஇ

**Comment by shirohana** [comment]
Should I insert `node_modules` paths start with `rootDir` to `module.paths` ?

like

```javascript
module.paths.splice(0, 0, join(this.options.rootDir, 'node_modules'))
```

When I works with `yarn link`, require.resolve does not looking for packages from the `rootDir`, and instead with

```javascript
[ 
// expect '/test-project/node_modules', here
  '/nuxt.js/dist/node_modules',
  '/nuxt.js/node_modules',
  '/node_modules'
]
```

which `/nuxt.js` is my Nuxt dir standalone with test project.

Maybe it would do nothing in almost Nuxt projects, but should I do it?

**Comment by pi0** [comment]
If `require.resolve` fails `resolvePath` should normally fallback to aliases which can start with `~` or `@`. So if the problem is only with linked packages we can use full paths. maybe mutating global and causing side-effects on Nuxt constructor resolve path isn't a good idea.

**Comment by pi0** [comment]
Another idea: We can add some logic into `resolvePath` that if normal resolve failed, try  `reuire.resolve(path.resolve(this.options.rootDir, 'node_modules', path))` before fallback to aliases.

**Comment by shirohana** [comment]
Ok (´・ω・｀)

Excuse me, is it a bad idea that test a module by linking to another project?

Should I try to write a test case instead?

**Comment by pi0** [comment]
@shirohana No and actually a good idea :)) Problem is with mutating global resolve path. Nuxt is running in some environments as a middleware too and it may cause unwanted bad side-effects.

**Comment by shirohana** [comment]
Understand (๑• ω •๑)

Is it acceptable by writing another function to resolve requires from `rootDir`?

By the way, I got a relative path from `relativeToBuild()` in `App.vue`

```javascript
import '../node_modules/bulma/bulma.sass'
```

Should it be related like that or just 'bulma/bulma.sass' like before better?

**Comment by pi0** [comment]
@shirohana For relative paths, yep seems good and consisten with other template paths, is there any problem you think would be with that?

> Is it acceptable by writing another function to resolve requires from rootDir?

Sure. But consider that function needs access to nuxt instance for `rootDir` (I mean can't be a pure utility). That would be greate if we extent core's `resolvePath` and adding explicit fallback for `rootDir` as mentioned in comments. I can help fix it if this approach looks OK with you too :)

**Comment by shirohana** [comment]
I was confused which is better between `../node_modules/bulma/bulma.sass` and `bulma/bulma.sass`, but now I'm not.

I'll try to extend the resolvePath function ヾ(\*´ω｀\*)ノ

**Comment by pi0** [comment]
Thanks for your time on contributing Nuxt.js :)

**Comment by shirohana** [comment]
I use node `module` to resolve requires from `rootDir`, is everything fine?

**Comment by pi0** [comment]
Nice seems working. I'll merge it BTW did not even found an unofficial article using `module` module 😆

**Comment by shirohana** [comment]
I found there's an obvious warning message caused by my commits today...

I'm sorry for that I didn't make a recheck at the time, let you fixed my mistake...

I'm sorry இωஇ

**Comment by pi0** [comment]
@shirohana Besides of that warning about external dependency is there any other warn? :)) BTW really thanks for helping us with Nuxt. Keep doing that :)

**Pull request merged**